hbase-server/src/main/java/org/apache/hadoop/hbase/master/AssignmentManager.java

```diff
@@ -253,6 +253,28 @@
   // are persisted in meta with a state store
   private final RegionStateStore regionStateStore;
 
+  /**
+   * Min version to consider for moving system tables regions to higher
+   * versioned RS. If RS has higher version than rest of the cluster but that
+   * version is less than this value, we should not move system table regions
+   * to that RS. If RS has higher version than rest of the cluster but that
+   * version is greater than or equal to this value, we should move system
+   * table regions to that RS. This is optional config and default value is
+   * empty string ({@link #DEFAULT_MIN_VERSION_MOVE_SYS_TABLES_CONFIG}).
+   * For instance, if we do not want meta region to be moved to RS with higher
+   * version until that version is >= 2.0.0, then we can configure
+   * "hbase.min.version.move.system.tables" as "2.0.0".
+   * When operator uses this config, it should be used with care, meaning
+   * we should be confident that even if user table regions come to RS with
+   * higher version (that rest of cluster), it would not cause any
```

Comment:
The language here is ambiguous. Better to say something like 

> When the operator uses this configuration option, any version between the current version and the new value of "hbase.min.version.move.system.tables" does not trigger any region movement. It is assumed the configured range of versions do not require special handling. 

This should also be committed to all branches, not just branch-1, for consistent functionality across all future releasing versions.

Reply:
@apurtell Created master branch PR #3439 yesterday. Will create branch-2 PR as well.